internal/store/bundle.go

```diff
@@ -175,6 +238,67 @@ func (b *bundleStore) storePath(ref reference.Reference) (string, error) {
 	return storeDir, nil
 }
 
+// Returns the list of all bundles present in the bundle store
+func (b *bundleStore) scanAllBundles() error {
+	digests := filepath.Join(b.path, "_ids")
+	if err := filepath.Walk(b.path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if !strings.HasSuffix(info.Name(), ".json") {
+			return nil
+		}
+
+		if strings.HasPrefix(path, digests) {
+			rel := path[len(digests)+1:]
+			dg := strings.Split(filepath.ToSlash(rel), "/")[0]
+			id := ID{digest.NewDigestFromEncoded(digest.SHA256, dg)}
+			appendRef(b.refsMap, id, id)
+			return nil
+		}
+
+		ref, err := b.pathToReference(path)
+		if err != nil {
+			return err
+		}
+		bndl, err := b.fetchBundleJSON(path)
+		if err != nil {
+			return err
+		}
+		id, err := FromBundle(bndl)
+		if err != nil {
+			return err
+		}
+		if _, ok := b.refsMap[id]; !ok {
+			b.refsMap[id] = []reference.Reference{ref}
+		} else {
+			b.refsMap[id] = append(b.refsMap[id], ref)
+		}
+
+		return nil
+	}); err != nil {
```

Comment:
I know it comes from a copy/past of previous code, but a `if` condition of 40 lines doesn't look really nice.
Maybe the walk call and the test on error can be separated like
```go
err := filepath.Walk(....)
if err != nil {
}
```
even if `if err := ...; err != nil {` is a go pattern

Or to be able to keep the pattern, maybe the walk function can be extracted so the code looks like:
```go
if err := filepath.Walk(b.path, walk); err != nil {
        return err
}
```

Reply:
I extracted the file processing function took by the directory walker into a separated function.